doc/source/development/contributing_codebase.rst

```diff
@@ -270,13 +270,17 @@ pandas uses `mypy <http://mypy-lang.org>`_ and `pyright <https://github.com/micr
 
 .. code-block:: shell
 
+    pre-commit run --hook-stage manual --all-files mypy
+    pre-commit run --hook-stage manual --all-files pyright
+    pre-commit run --hook-stage manual --all-files pyright_reportGeneralTypeIssues
     # the following might fail if the installed pandas version does not correspond to your local git version
-    pre-commit run --hook-stage manual --all-files
+    pre-commit run --hook-stage manual --all-files stubtest
 
-    # if the above fails due to stubtest
-    SKIP=stubtest pre-commit run --hook-stage manual --all-files
+in your active python environment.
 
-in your activated python environment. A recent version of ``numpy`` (>=1.22.0) is required for type validation.
+.. warning::
+
+    * It is essential that the versions of ``mypy`` and ``numpy`` in the active environment are the same as in ``requirements-dev.txt``.
```

Comment:
There is no version number of `numpy` in `requirements-dev.txt` or `environment.yml` . 

I think we should be adding a statement about updating the environment.   For conda environments, you need to de-activate the environment, and do `mamba env update -f environment.yml` .  Not sure how to do the updates when using `pip`, so we should indicate both ways.  See my comment here:  https://github.com/pandas-dev/pandas/issues/51313#issuecomment-1427056543

Reply:
I added a link to the python setup docs.

> What I've found is that you have to stage your changes via git add in order for the commands you provided to work. 

When adding  `--all-files` this does not seem to be needed. Without it pre-commit says:

> [WARNING] Unstaged files detected.
> [INFO] Stashing unstaged files to /var/home/twoertwein/.cache/pre-commit/patch1676322301-669741.

Reply:
> When adding `--all-files` this does not seem to be needed. Without it pre-commit says:

Hmmm.  Maybe I forgot to include that.